ReactAndroid/src/main/java/com/facebook/react/devsupport/DevLoadingView.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.devsupport;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.view.View;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import com.facebook.common.logging.FLog;
+import com.facebook.react.R;
+import com.facebook.react.bridge.UiThreadUtil;
+import com.facebook.react.common.ReactConstants;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+/**
+ * View to display loading messages on top of the screen. All methods are thread safe.
+ */
+public class DevLoadingView extends RelativeLayout {
+  private static boolean sEnabled = true;
+
+  public static void setDevLoadingEnabled(boolean enabled) {
+    sEnabled = enabled;
+  }
```

Comment:
iOS had this feature so I added here too. This will be useful for example in Exponent to be able to show a custom loading screen.